src/main/java/nextstep/qna/service/QnAService.java

```diff
@@ -25,25 +23,42 @@ public class QnAService {
 
     @Transactional
     public void deleteQuestion(NsUser loginUser, long questionId) throws CannotDeleteException {
-        Question question = questionRepository.findById(questionId).orElseThrow(NotFoundException::new);
+        Question question = findQuestionByQuestionId(questionId);
+        validateIfLoginUserHasAuthorityToDeleteQuestion(loginUser, question);
+
+        question.answersCanBeDeleted(loginUser);
+
+        question.delete(true);
+        addQuestionToDeleteHistories(question);
+
+        deleteAnswersAndAddThemToDeleteHistories(question);
+        deleteHistoryService.saveAll();
```

Comment:
```suggestion
         deleteHistoryRepository.saveAll(toHistory(question));
```
```java
    private static DeleteHistory toHistory(Question question) {
        return new DeleteHistory(ContentType.QUESTION, question.getId(), question.getWriter(), LocalDateTime.now());
    }
```

- 위 로직은 이해가 잘 안가는데요. deleteHistory에 매번 추가한 후 saveAll을 하면, 이전 기록들이 중복해서 들어가게 되지 않나요? 위와 같이 작성하는건 어떨까요?